scripts/update_versions.py

```diff
@@ -0,0 +1,163 @@
+
+def update_dependencies(mod_path, dependencies):
+    # Open the go.mod file in read mode
+    with open(mod_path + "/go.mod", 'r') as mod_file:
+        # Read the contents of the file into a list of lines
+        lines = mod_file.readlines()
+    changed = False
+    # Loop through the lines of the go.mod file
+    for i, line in enumerate(lines):
+        # Check if the line starts with "require"
+        # Split the line by spaces
+        parts = line.split(' ')
+        if len(parts) < 2: # not a require line
+            continue
+        # Get the dependency name and version from the line
+        name = parts[0].strip()
+        version = parts[1].strip()
+        rem_parts = ' '.join(parts[2:]).strip()
+        # Check if the dependency name is in the dictionary of desired versions
+        if name in dependencies:
+            human_version = version
+            if "-" in human_version:
+                human_version = version.split("-")[0]
+            new_version = human_version + "-" + dependencies[name]
+            # Update the version in the go.mod file
+            print(lines[i])
+            print(dependencies[name])
+            lines[i] = f"\t{name} {new_version} {rem_parts}\n"
+            print(lines[i])
+            changed = True
+
+    if not changed:
+        return
+    # Open the go.mod file in write mode
+    with open(mod_path + "/go.mod", 'w') as mod_file:
+        # Write the modified lines to the file
+        mod_file.writelines(lines)
+
+def get_go_sum_version():
+  # Define the command as a list of strings
+  command = "git --no-pager show --quiet --abbrev=12 --date=format-local:%Y%m%d%H%M%S --format=%cd-%h".split(" ")
+  # print(command)
+  env = {'TZ': 'UTC'}
+
+  # Run the command
+  result = subprocess.run(command, env=env, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+  # Print the output
+  return result.stdout.decode('utf-8').strip()
+
+def check_for_diff_against_main(path):
+  # Run the 'git diff' command and store the output in a variable
+  output = subprocess.run(['git', 'diff', 'main', "--exit-code", '--', path], stdout=subprocess.PIPE)
+  # Check the exit status of the command (!- 0 means there was a diff)
+  return output.returncode != 0
+
+def check_for_diff_since_last_commit():
+  output = subprocess.run(['git', 'diff', "--exit-code"], stdout=subprocess.PIPE)
+  return output.returncode != 0
+
+def build_di_graph(names, mod_filepaths):
+  # Create an empty directed graph
+  G = nx.DiGraph() 
+
+  # Iterate over the list of go.mod filepaths
+  for filepath in mod_filepaths:
+    curname = parse_go_mod_name(filepath)
+    # Open the go.mod file in read mode
+    with open(filepath + "/go.mod", 'r') as mod_file:
+      # Read the file contents into a string
+      contents = mod_file.read()
+      
+      # Use a regular expression to match the 'require' block and extract the dependencies
+      dependencies = re.findall(r'require\s+\(\s*(.*?)\s*\)', contents, re.DOTALL)      
```

Comment:
This assumes no replace directives, which I think is safe to assume.

Reply:
```shell
$ find . -name go.mod | xargs grep replace
./go.mod:replace (
./go.mod:       // ToDo: replace the commit hash with v0.30.0-osmo-v14 once the version is tagged
./osmomath/go.mod:replace (
./x/ibc-hooks/go.mod:replace (
./osmoutils/go.mod:replace (
```